File: examples/python/pink-about.py
#!/usr/bin/env python
# coding: utf-8
# vim: set sw=4 ts=4 sts=4 et :

"""
Simple example showing how to use PinkTrace version constants
"""

from __future__ import print_function

import pinktrace

print("Built using PinkTrace %d.%d.%d%s" % (pinktrace.VERSION_MAJOR,
    pinktrace.VERSION_MINOR, pinktrace.VERSION_MICRO,
    pinktrace.VERSION_SUFFIX), end="")

if pinktrace.GIT_HEAD:
    print(" git-%s" % pinktrace.GIT_HEAD)
else:
    print()

File: examples/ruby/pink-about.rb
#!/usr/bin/env ruby
# coding: utf-8
# vim: set sw=2 sts=2 et fenc=utf-8 :

=begin
Simple example showing how to use PinkTrace version constants
=end

require 'PinkTrace'

print 'Built using ', PinkTrace::PACKAGE, " ",
  PinkTrace::VERSION_MAJOR, ".",
  PinkTrace::VERSION_MINOR, ".",
  PinkTrace::VERSION_MICRO,
  PinkTrace::VERSION_SUFFIX

unless PinkTrace::GIT_HEAD.empty?
  print ' git-', PinkTrace::GIT_HEAD
end

puts
